control_node:
  ros__parameters:

    protocol: "usb"
    serial_port: "/dev/ttyACM0"

    publish_rate: 50.0

    cmd_vel_topic: "cmd_vel"
    odom_topic: "wheel/odom"
    joint_states_topic: "joint_states"

    wheel_separation: 0.788
    wheel_radius: 0.11736
    wheel_reduction: 100.0
    wheel_encoder_cpr: 64

    pid_kp: 1.0
    pid_ki: 0.01
    pid_kd: 0.0001

    pid_p_bound: 100.0
    pid_i_bound: 50.0
    pid_d_bound: 50.0

    wheel_joint_names: ["front_left_wheel_joint", "front_right_wheel_joint", "back_left_wheel_joint", "back_right_wheel_joint"]

    pose_covariance_diagonal: [0.002, 0.002, 0.0, 0.0, 0.0, 0.02]
    twist_covariance_diagonal: [0.002, 0.0, 0.0, 0.0, 0.0, 0.02]

    base_frame_id: base_link
    odom_frame_id: odom

    is_open_loop: true

    cmd_vel_timeout: 0.5

    linear.x.has_velocity_limits: true
    #linear.x.has_acceleration_limits: true
    # linear.x.has_jerk_limits: false
    linear.x.max_velocity: 2.0
    linear.x.min_velocity: -2.0
    #linear.x.max_acceleration: 5.0
    #linear.x.min_acceleration: -5.0
    # linear.x.max_jerk: NAN
    # linear.x.min_jerk: NAN

    angular.z.has_velocity_limits: true
    #angular.z.has_acceleration_limits: true
    # angular.z.has_jerk_limits: false
    angular.z.max_velocity: 2.50
    angular.z.min_velocity: -2.50
    #angular.z.max_acceleration: 5.0
    #angular.z.min_acceleration: -5.0
    # angular.z.max_jerk: NAN
    # angular.z.min_jerk: NAN

